modules/react-native-status/ios/RCTStatus/Jail.m

```diff
@@ -0,0 +1,234 @@
+    
+    return result;
+}
+
+- (NSDictionary *)call:(NSString *)chatId
+                  path:(NSString *)path
+                params:(NSString *)params
+{
+    Cell *cell = [_cells valueForKey:chatId];
+    JSContext *context = cell.context;
+    if(cell == nil) {
+        return [[NSDictionary alloc] initWithObjectsAndKeys:nil, @"result", @"jail is not initialized", @"error", nil];
+    }
+    JSValue *callResult = [context[@"call"] callWithArguments:@[path, params]];
+    JSValue *exception = [context exception];
+    NSString *error;
+    if(exception != nil) {
+        error = [exception toString];
+        [context setException:nil];
+    }
+    NSDictionary *result = [[NSDictionary alloc] initWithObjectsAndKeys:[callResult toString], @"result", error, @"error", nil];
+    
+    return result;
+}
+
+- (Cell *)createCell:(NSString *)chatId
+                 withCode:(NSString *)js
+{
+    if(_cells == nil) {
+        _cells = [NSMutableDictionary dictionaryWithCapacity:1];
+    }
+    
+    Cell * cell = [Cell new];
+    JSContext *context = [JSContext new];
+    cell.context = context;
+    
+    HandlersJs *handlers = [HandlersJs new];
+    [handlers setCahtId:chatId];
+    [handlers addToContext:context];
+    
+    [self addTimer:cell];
+    [context evaluateScript:_initialJs];
+    
+    JSValue *excep = [context exception];
+    NSLog(@"err1 %@", [excep toString]);
+    
+    NSString *webJs =
+    NSStringMultiline
+    (
+     var statusSignals = {
+     sendSignal: function (s) {statusNativeHandlers.sendSignal(s);}
+     };
+     var setTimeout = function (fn, t) {
+         var args = [fn, t];
+         var getItem = function(idx) {return args[idx];};
+         return jsTimer.setTimeout(getItem);
+     };
+     var setInterval = function (fn, t) {
+         var args = [fn, t];
+         var getItem = function(idx) {return args[idx];};
+         return jsTimer.setInterval(getItem);
+     };
+     var clearInterval = function (id) {
+         jsTimer.clearInterval(id);
+     };
+     var Web3 = require('web3');
+     var provider = {
+     send: function (payload) {
+         var result = statusNativeHandlers.send(JSON.stringify(payload));
+         return JSON.parse(result);
+     },
+     sendAsync: function (payload, callback) {
+         var wrappedCallback = function (result) {
+             console.log(result);
+             var error = null;
+             try {
+                 result = JSON.parse(result);
+             } catch (e) {
+                 error = result;
+             }
+             callback(error, result);
+         };
+         var args = [JSON.stringify(payload), wrappedCallback];
+         var getItem = function(idx) {return args[idx];};
+         statusNativeHandlers.sendAsync(getItem);
+     }
+     };
+     var web3 = new Web3(provider);
+     var console = {
+     log: function (data) {
+         statusNativeHandlers.log(data);
+     }
+     };
+     var Bignumber = require("bignumber.js");
```

Comment:
Can't we inject web3.js here? Why are we doing it in cljs? I see we are doing something like this above but confused why we are doing `(str status js-res/web3)` as well